### PR TITLE
Feature/id 23/api gateway

### PR DESCRIPTION
**Pull request by Elvis-Castro-H**
## Pull Request
**Clickup:** https://app.clickup.com/t/868ac266e

**Note:** This implementation does not include API registration in the service registry

## Reviewers
@victor-fundacion 
@JP-Arequipa

### CODE REVIEW

**Review by victor-fundacion — CHANGES_REQUESTED** [review]
Resolve merge conflicts

**Review by victor-fundacion — CHANGES_REQUESTED** [review]
Create Readme on how to start the api and test it with other services

File: apps/docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,71 @@
...
+#      - servicediscovery
+#      - sqlserver
+#    networks:
+#      - shared-network
+
+  # Service Discovery
+  servicediscovery:
+    build:
+      context: ./ServiceDiscovery
+      dockerfile: Dockerfile
```
Comment (victor-fundacion) [inline apps/docker-compose.yml:37]: Service Discovery can not be started.
Screenshot from 2024-10-30 10-46-44

**Review by victor-fundacion — CHANGES_REQUESTED** [review]
I followed the steps on each Readme and could not register a service in service discovery.
1.  I run docker compose (to init sql and consul)
image
2. Started the service discovery service manually
image
3. Started the CEventservice.api service and gateway
4. Made a POST request to `POST` request to `/api/service-registry/register`
with:
{
  "serviceName": "cevent-service",
  "address": "localhost",
  "port": 5000,
  "healthCheckEndpoint": "/health"
}

Got this error:
image

Consul up:
image

**Comment by victor-fundacion** [comment]
@Elvis-Castro-H Add git ignore to your apigateway and servicediscovery

**Comment by Elvis-Castro-H** [comment]
@victor-fundacion I already added a .gitignore file in each service implemented, I also added a general .gitignore in the root of apps

**Review by victor-fundacion — APPROVED** [review]
Observations solved and apigatway working correctly.
image

**Review by JP-Arequipa — APPROVED** [review]

**Pull request merged**